src/iris/sender/rpc.py

```diff
@@ -129,10 +143,13 @@ def handle_api_notification_request(socket, address, req):
                        address, role, target, notification['application'],
                        notification.get('priority', notification.get('mode', '?')))
 
-    for _target in expanded_targets:
-        temp_notification = notification.copy()
-        temp_notification['target'] = _target
-        send_funcs['message_send_enqueue'](temp_notification)
+    if role == 'literal_target':
```

Comment:
The `literal_target` pseudo-role is fine for interfacing with users, but once we drop into the API layer we should disambiguate from real roles. I'd prefer an additional member in the notification dict of `unexpanded: True` or something. This way `role` in the rpc and sender code always means a real role.